pyrit/orchestrator/fuzzer_orchestrator.py

```diff
@@ -0,0 +1,414 @@
+        *,
+        prompts: list[str],   #questions in GPTFuzzer
+        prompt_target: PromptTarget,
+        prompt_templates: list[str], # list of all the jailbreak prompts on which MCTS will be applied.
+        prompt_converters: Optional[list[PromptConverter]] = None,
+        template_converter:list[PromptConverter], # shorten/expand 
+        scorer: Scorer,
+        memory: Optional[MemoryInterface] = None,
+        memory_labels: Optional[dict[str, str]] = None,
+        verbose: bool = False,
+        frequency_weight=0.5, reward_penalty=0.1, minimum_reward=0.2,
+        non_leaf_nodeprobability =0.1,
+        random_state = None,
+        batch_size: int = 10,
+    ) -> None:
+
+        """Creates an orchestrator that explores a variety of jailbreak options via fuzzing.
+        
+        Paper: GPTFUZZER: Red Teaming Large Language Models with Auto-Generated Jailbreak Prompts.
+
+            Link: https://arxiv.org/pdf/2309.10253
+
+            Author: Jiahao Yu, Xingwei Lin, Zheng Yu, Xinyu Xing
+
+            GitHub: https://github.com/sherdencooper/GPTFuzz/tree/master
+        
+        Args:
+        
+            prompts: The prompts will be the questions to the target. 
+            
+            prompt_target: The target to send the prompts to.
+            
+            prompt_templates: List of all the jailbreak templates which will act as the seed pool. 
+            At each iteration, a seed will be selected using the MCTS-explore algorithm which will be sent to the 
+            shorten/expand prompt converter.
+            
+                          The converted template along with the prompt will be sent to the target. 
+                          
+            prompt_converters: The prompt_converters to use to convert the prompts before sending 
+            them to the prompt target. 
+            
+            template_converter: The converter that will be applied on the jailbreak template that was 
+            selected by MCTS-explore. 
+            
+                          The prompt converters will not be applied to the prompts. 
+                          Shorten/expand prompt converters.
+                          
+            scorer: The scorer classifies the prompt target outputs as sufficient (True) or insufficient (False)
+                to satisfy the objective that is specified in the prompt.
+                
+            memory: The memory to use to store the chat messages. If not provided, a DuckDBMemory 
+            will be used.
+            
+            memory_labels: The labels to use for the memory. This is useful to identify the messages in the memory.
+            
+            verbose: Whether to print debug information.
+
+            frequency_weight: constant that balances between the seed with high reward and the seed that is selected fewer times.
+
+            Reward penalty: Reward penalty diminishes the reward for the current node and its ancestors when the path lengthens.

```

Comment:
```suggestion
            reward_penalty: Reward penalty diminishes the reward for the current node and its ancestors when the path lengthens.
```
Same with others. They should be consistent with arg names